components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubHubAdapterServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.identity.event.common.publisher.model.EventContext;
+import org.wso2.identity.event.common.publisher.EventPublisher;
+import org.wso2.identity.event.websubhub.publisher.internal.WebSubHubAdapterDataHolder;
+import org.wso2.identity.event.common.publisher.model.SecurityEventTokenPayload;
+import org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants;
+import org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil;
+import org.wso2.identity.event.websubhub.publisher.exception.WebSubAdapterException;
+
+import java.io.IOException;
+
+import static org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil.handleServerException;
+
+/**
+ * OSGi service for publishing events using web sub hub.
+ */
+public class WebSubHubAdapterServiceImpl implements EventPublisher {
+
+    private static final Log log = LogFactory.getLog(WebSubHubAdapterServiceImpl.class);
+    private String webSubHubBaseUrl = null;
+
+    @Override
+    public void publish(SecurityEventTokenPayload eventPayload, EventContext eventContext, String eventUri)
+            throws WebSubAdapterException {
+
+        log.info("Event published with new generic component.");
+        if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().isAdapterEnabled()) {
+            WebSubHubAdapterUtil.makeAsyncAPICall(eventPayload, eventContext, constructHubTopic(eventUri,
+                            eventContext.getTenantDomain()),
+                    getWebSubBaseURL());
```

Comment:
So here we seems not be returning any state.
Ideally we should have troubleshooting diagnostic logs on whether the event was published or not
If we don't return a state from the implementation, per adapter we have to handle it. Or we can move the logger to the top layer

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/7e62e79fe00e7222402c1e7508d0ce4d065b24d0